core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/PartialSuccessHandler.java

```diff
@@ -0,0 +1,196 @@
+ * @author jamdavi
+ *
+ */
+public class PartialSuccessHandler implements TransmissionHandler {
+
+	/**
+	 * Ctor
+	 * 
+	 * Constructs the PartialSuccessHandler object.
+	 * 
+	 * @param policy
+	 *            The {@link TransmissionPolicyManager} object that is needed to
+	 *            control the back off policy.
+	 */
+	public PartialSuccessHandler(TransmissionPolicyManager policy) {
+	}
+
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+	}
+
+	/**
+	 * Provides the core logic for the retransmission
+	 * 
+	 * @param args
+	 *            The {@link TransmissionHandlerArgs} for this transmission.
+	 * @return Returns a pass/fail for handling this transmission.
+	 */
+	boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
+		if (args.getTransmission() != null && args.getTransmissionDispatcher() != null) {
+			switch (args.getResponseCode()) {
+			case HttpStatus.SC_PARTIAL_CONTENT:
+				BackendResponse beR = getBackendResponse(args.getResponseBody());
+				List<String> originalItems = generateOriginalItems(args);
+
+				// Somehow the amount of items received and the items sent do not match
+				if (beR != null && (originalItems.size() != beR.itemsReceived)) {
+					InternalLogger.INSTANCE.trace(
+							"Skipping partial content handler due to itemsReceived being larger than the items sent.");
+					return false;
+				}
+
+				if (beR != null && (beR.itemsAccepted != beR.itemsReceived)) {
+
+					List<String> newTransmission = new ArrayList<String>();
+					for (BackendResponse.Error e : beR.errors) {
+						switch (e.statusCode) {
+						case HttpStatus.SC_REQUEST_TIMEOUT:
+						case HttpStatus.SC_INTERNAL_SERVER_ERROR:
+						case HttpStatus.SC_SERVICE_UNAVAILABLE:
+						case 429:
+						case 439:
+							// Unknown condition where backend response returns an index greater than the
+							// items we're returning
+							if (e.index < originalItems.size()) {
+								newTransmission.add(originalItems.get(e.index));
+							}
+							break;
+						}
+					}
+					return sendNewTransmission(args, newTransmission);
+				}
+				InternalLogger.INSTANCE
+						.trace("Skipping partial content handler due to itemsAccepted and itemsReceived being equal.");
+				return false;
+
+			default:
+				InternalLogger.INSTANCE.trace("Http response code %s not handled by %s", args.getResponseCode(),
+						this.getClass().getName());
+				return false;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Used to parse the original telemetry request in order to resend the failed
+	 * ones.
+	 * 
+	 * @param args
+	 *            The {@link TransmissionHandlerArgs} that contains the
+	 *            {@link Transmission} object.
+	 * @return A List<> of each sent item
+	 */
+	List<String> generateOriginalItems(TransmissionHandlerArgs args) {
+		List<String> originalItems = new ArrayList<String>();
```

Comment:
Types in <> in Java 7 and onwards can be auto-inferred and is more efficient.  This can be replaced with 
List<String> originalItems = new ArrayList<>();